Competitive Programming/c2cpp.md

```diff
@@ -0,0 +1,117 @@
+# C to Cpp for competitive
+
+This article is written for specifically keeping competetive in mind and not for using cpp in development or any other context. Also this aims at getting started to use good and important features of c++ as quickly as possible so skips many details.
+Most of the c++ fetures talked here work on c++98 and after but some work only after c++11 so for simplicity we will assume to be using c++11. Also we will be using gcc/g++ compiler. You can also use an IDE like codeblocks it already has gcc and g++ compilers by default (MINGW); remember to enable c++11 or c++14 option in codebloks' settings.
+
+Once you are familiar with C its easy to migrate to and learn C++. First of all since c++ is superset of C so you can use most of the concepts of c in c++. So we will first talk about porting code and then a couple of things new in c++ useful for competetive.
+
+# PREREQUISITES
+We require that you are atleast aware of the following concepts in c:
+
+* Basic Syntax of C (input, output, and logical, relational, arithmetic & bitwise operators):  
+* Loops
+* Arrays/Strings
+* Functions
+* Structures
+* Pointers
+
+If you don't know any of the above things, its better to have a recap first and than come back to this article.
+Also we WON'T require things like enums, unions, static, file IO etc. so as long as you are familiar with the ones above you are good to go.
+
+# PORTING C CODE
+So all of the things listed in prerequisites remain same in C++. You can use the same `scanf` and `printf` in C++ also; they are defined in header files <stdio.h> and <cstdio>. Similarly string functions are defined in header files <string.h> and <cstring>, math functions in math.h and <cmath>. Though the files listed in pairs are same for most intend and purposes but have a subtle difference which requires knowledge of namespaces. Though we will not go into this discussion but I have highlighted a few recources at the end of this section for any qurious mind. Though NOT recommended for beginners.
```

Comment:
I think it'll be better to tell people to only use `<cxxx>` headers with a `using namespace std;` instead of `<xxx.h>` headers.